ui/components/data-sources/AddDataSourceDialog.js

```diff
@@ -0,0 +1,216 @@
+                <Modal.Header>
+                    <button
+                        className="close"
+                        aria-hidden="true"
+                        onClick={() => this.onClose()}
+                        aria-label="Close"
+                        type="submit"
+                    >
+                        <Icon type="pf" name="close" />
+                    </button>
+                    <Modal.Title>Add Data Source</Modal.Title>
+                </Modal.Header>
+
+                <Modal.Body>
+                    {/* Alert */}
+                    {err && <Alert onDismiss={() => this.setState({ err: "" })}>{err}</Alert>}
+
+                    <Form horizontal>
+
+                        {/* Data Source Name */}
+                        <FormGroup controlId="name" validationState={validations.name}>
+                            <Col sm={3}>Data Source Name</Col>
+                            <Col sm={9}>
+                                <FormControl
+                                    type="text"
+                                    value={name}
+                                    onChange={e => this.onNameChange(e.target.value)}
+                                />
+                            </Col>
+                        </FormGroup>
+
+                        {/* Data Source Type */}
+                        <FormGroup controlId="type" validationState={validations.type}>
+                            <Col sm={3}>Data Source Type</Col>
+                            <Col sm={9}>
+                                <InputGroup>
+                                    <FormControl
+                                        disabled
+                                        style={{ backgroundColor: "#fff", color: "#363636" }}
+                                        value={type}
+                                    />
+                                    <InputGroup.Button>
+                                        <DropdownButton
+                                            bsStyle="default"
+                                            id="dropdown-type"
+                                            title=""
+                                            onSelect={key => this.onTypeChange(key)}
+                                        >
+                                            <MenuItem eventKey={DataSourceType.InMemory}>
+                                                {DataSourceType.InMemory}
+                                            </MenuItem>
+                                            {/* More Data Source Types to be added */}
+                                        </DropdownButton>
+                                    </InputGroup.Button>
+                                </InputGroup>
+                            </Col>
+                        </FormGroup>
+
+                        {/* Config */}
+                        <FormGroup controlId="config" validationState={validations.config}>
```

Comment:
Is the idea here to let the user provide a JSON string with the config? How would that work for the in-memory data source, something like `{"collection": "my_test_collection"}`? Would it not make more sense to let the user input the collection name and then deal with converting it to JSON on the backend?